src/coreclr/jit/block.h

```diff
@@ -714,6 +714,31 @@ struct BasicBlock : private LIR::Range
 
     BasicBlockFlags bbFlags;
 
+    bool CheckFlag(const BasicBlockFlags flag, const BasicBlockFlags target) const
+    {
+        return ((bbFlags & flag) == target);
+    }
+
+    bool CheckFlag(const BasicBlockFlags flag) const
+    {
+        return CheckFlag(flag, flag);
+    }
+
+    bool HasFlag(const BasicBlockFlags flag) const
```

Comment:
Should this assert that `flag` has only one bit set? And add an explicit `HasAnyFlag` that allows for the rare cases where you're checking if any flag is set? (actually, we should probably just make callers use `HasFlag(one) || HasFlag(two)` in that case)

Reply:
I think that's a good idea. It doesn't come up often, but we do have a few `HasFlag(BBF_FLAG1 | BBF_FLAG2)` calls, so maybe we can use the templated approach we use for `BasicBlock::KindIs`? If not, we can just use more than one call to `HasFlag` in the caller.

Reply:
I think it's better to make the caller use `HasFlag(one) || HasFlag(two)`, or `HasFlag(one) && HasFlag(two)`.

Yes, we could use the template magic to have `HasAnyFlags(one,two,...)` and `HasAllFlags(one,two,...)` but does that happen very often? IMO, we should not have `HasFlag(one,two,...)` because it's not clear if it's "any" or "all".

Reply:
> Yes, we could use the template magic to have HasAnyFlags(one,two,...) and HasAllFlags(one,two,...) but does that happen very often?

I don't recall seeing it enough to justify introducing two more methods.

> I think it's better to make the caller use HasFlag(one) || HasFlag(two), or HasFlag(one) && HasFlag(two).

Now that you mention the `||` vs `&&` cases, I agree it's more readable to just have the caller make two calls.

Reply:
On second thought, there are some cases where multiple calls to `HasFlag` quickly becomes ugly, so I'll introduce `HasAnyFlags` to cover these scenarios (I haven't found a need for `HasAllFlags` anywhere).

Reply:
nit: in English, the preferred wording would be HasAnyFlag and HasAllFlags, IMO